cmd/bundle/run.go

```diff
@@ -150,8 +150,6 @@ Example usage:
 		}
 
 		diags = diags.Extend(bundle.ApplySeq(ctx, b,
-			terraform.Interpolate(),
-			terraform.Write(),
```

Comment:
When code calls into TF, we need the `bundle.tf.json` file to be present for the provider definition.

I believe that load does not _have_ to call into TF, but now it still does with an initialization call.

If you run this command with an empty cache, I expect it to fail because it won't have any configuration.

Reply:
> When code calls into TF, we need the bundle.tf.json file to be present for the provider definition.

could you make it more specific? which mutator in run/open commands calls into TF? I don't see where they do plan or apply.

Reply:
I looked more into this:

- terraform.Interpolate: modifies bundle in-place references like [resources.jobs.foo.id](http://resources.jobs.foo.id/) are replaced with tf-specific references
- terraform.Write: convert bundle config to terraform and writes bundle.tf.json

None of the follow up mutators depend on the above:

- statemgmt.StatePull() downloads state from remote
- statemgmt.Load() merges ids from state file into bundle config and updates modified_status as well
- bundle/run/* does not depend on terraform

We have a bunch of tests in acceptance/bundle/run

Reply:
@denik not exactly. There might be cases when bundle deploy is done from one place / pipeline and then bundle run for the same bundle is done from another place / pipeline. No local state but only remote one. I don;t think there's a test like this

Reply:
Newer configuration that is not deployed is not going to have IDs for most resources, so I don't see why.

Here's a test that wipes out state to see that it is pulled from remote: 61cb3df8bd47bd239bdc6ae3f22024b6f886989c